src/main/scala/org/graphframes/GraphFrame.scala

```diff
@@ -482,11 +480,52 @@ class GraphFrame private(
    */
   private[graphframes] lazy val indexedEdges: DataFrame = {
     val packedEdges = edges.select(col(SRC), col(DST), nestAsCol(edges, ATTR))
-    val indexedSourceEdges = packedEdges
-      .join(indexedVertices.select(col(LONG_ID).as(LONG_SRC), col(ID).as(SRC)), SRC)
-    val indexedEdges = indexedSourceEdges.select(SRC, LONG_SRC, DST, ATTR)
-      .join(indexedVertices.select(col(LONG_ID).as(LONG_DST), col(ID).as(DST)), DST)
-    indexedEdges.select(SRC, LONG_SRC, DST, LONG_DST, ATTR)
+    if (hasIntegralIdType) {
+      packedEdges.select(
+        col(SRC), col(SRC).cast("long").as(LONG_SRC),
```

Comment:
Nevermind, I just saw your other PR.  It was originally intentional to support fewer bits in the ID, but using Long everywhere seems fine.